### PR TITLE
Update `main.yml` workflow

### PR DESCRIPTION
**Pull request by rikhuijzer**
This PR suggests to update Hugo to a (much) newer Hugo version. This fixes an issue where the sentence
```
_Raw Buffer Floating-point Atomic Add (MI-* only)_
```
is not parsed as an italicized sentence, see <https://mlir.llvm.org/docs/Dialects/AMDGPU/#amdgpuraw_buffer_atomic_fadd-mliramdgpurawbufferatomicfaddop>. This came up in https://reviews.llvm.org/D152621.

A preview of the website generated with Hugo 0.111 is available at <https://rikhuijzer.github.io/mlir-www>. Note that any relative link like `/foo/bar.png` is broken on that website because I'm serving the preview at `/mlir-www/...` instead of `/...`. Apart from those links, I am not able to find any issues related to updating the Hugo version.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -11,18 +11,20 @@ on:
 
 jobs:
   build-deploy:
+    permissions:
+      contents: write
```
Comment (rikhuijzer) [inline .github/workflows/main.yml:15]: This ensures that `secrets.GITHUB_TOKEN` has permission to write to a secondary branch. This allows other people to fork `mlir-www` and have a working workflow. See, for example, https://github.com/JuliaCI/PkgTemplates.jl/pull/347 for more information about this.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -59,8 +61,9 @@ jobs:
 
     - name: Deploy
       uses: peaceiris/actions-gh-pages@v3
+      if: github.ref == 'refs/heads/main'
```
Comment (rikhuijzer) [inline .github/workflows/main.yml:64]: This ensures that the pages are only published when a workflow is triggered by a change to `main`. Copied from <https://github.com/peaceiris/actions-gh-pages>.

**Review by jpienaar — APPROVED** [review]
Thanks, didn't do any exhaustive checking but it looked good from random sampling.

File: .github/workflows/main.yml (hunk 1 above)
Comment (joker-eph) [reply .github/workflows/main.yml:15]: Can you add a comment for this in the file itself?

**Pull request merged**